domain/src/main/java/org/oppia/domain/oppialogger/crashlytics/ExceptionsController.kt

```diff
@@ -0,0 +1,174 @@
+package org.oppia.domain.oppialogger.crashlytics
+
+import androidx.lifecycle.LiveData
+import org.oppia.app.model.ExceptionLog
+import org.oppia.app.model.OppiaExceptionLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.domain.oppialogger.ExceptionLogStorageCacheSize
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.ConsoleLogger
+import org.oppia.util.logging.ExceptionLogger
+import org.oppia.util.networking.NetworkConnectionUtil
+import org.oppia.util.system.OppiaClock
+import javax.inject.Inject
+
+class ExceptionsController @Inject constructor(
+  private val exceptionLogger: ExceptionLogger,
+  private val dataProviders: DataProviders,
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val consoleLogger: ConsoleLogger,
+  private val networkConnectionUtil: NetworkConnectionUtil,
+  private val oppiaClock: OppiaClock,
+  @ExceptionLogStorageCacheSize private val exceptionLogStorageCacheSize: Int
+) {
+  private val exceptionLogStore =
+    cacheStoreFactory.create("exception_logs", OppiaExceptionLogs.getDefaultInstance())
+
+  lateinit var exceptionCause: Throwable
+
+  fun logException(exception: Exception, timestamp: Long) {
```

Comment:
Why not always log this from within our ``Logger``class? I don't think callsites should be deciding whether to care about caching/uploading exception logs. Instead, callsites just care about noting to developers that something went wrong, and after that it's up to the logging system to decide what to do with the exception.

Reply:
I think this will be done during restructuring as the `oppiaLogger` will become the call site for the developers. Please correct me if I am wrong @vinitamurthi.

Reply:
Yeah we will do this in the OppiaLogger